### ISSUE TITLE
fromPromise.case typings is wrong

### ISSUE THREAD

**Issue by alvteren** [mobxjs/mobx-utils#210]
You have:
```ts
export type IBasePromiseBasedObservable<T> = {
    isPromiseBasedObservable: true
    case<U>(handlers: { pending?: () => U; fulfilled?: (t: T) => U; rejected?: (e: any) => U }, defaultFulfilled?: boolean): U
} & PromiseLike<T>
```
You should have:
```ts
export type IBasePromiseBasedObservable<T> = {
    isPromiseBasedObservable: true
    case<U>(handlers: { pending?: (t?: T) => U; fulfilled?: (t: T) => U; rejected?: (e: any) => U }, defaultFulfilled?: boolean): U
} & PromiseLike<T>
```
Will replace: 
`pending?: () => U` -> `pending?: (t?: T) => U`

**Comment by ItamarShDev** [comment]
Feel free to open PR

**Comment by thbt** [comment]
This should be fixed by #208
